sklearn/_config.py

```diff
@@ -93,6 +95,12 @@ def set_config(
 
         .. versionadded:: 1.1
 
+    use_pairwise_dist : bool, default=None
```

Comment:
Is this for testing purposes or something we intend for users?

Reply:
This is solely for testing purpose. This is the simplest way I found to test both back-ends for `'brute'` via `test_kneighbors_brute_backend`.

To me `use_pairwise_dist` and `pairwise_dist_chunk_size` must not be part of the configuration of scikit-learn.

Reply:
I think it can be useful for maintainers to quickly check the impact of the new Cython implementation. But this should be made explicit in the docstring.

"""
Intended for easier benchmarking and testing of scikit-learn internals. End users are not
expected to benefit from customizing this configuration setting.
"""

Also I would suggest to rename this to `enable_cython_pairwise_dist=True/False` or `pairwise_dist_impl="cython"/"numpy_scipy"`  to make it more explicit that this is a switch between the new Cython/OpenMP implementation or the old numpy/scipy-based implementation.

I think I slightly prefer the second option.